core/src/gl/uniformEntries.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <string>
+
+namespace Tangram {
+
+namespace UniformEntries {
+
+typedef uint16_t EntryId;
+
+struct UniformEntry {
```

Comment:
Seems to me like we don't need the ID/index/handle to be co-located with the name, since the name only needs to be accessed rather infrequently. We could keep a separate, global array of name strings that we index into with the IDs.